### ISSUE TITLE
Can someone help me understand this error?

### ISSUE THREAD

**Issue by Smackdab69** [sim0n00ps/OF-DL#664]
I sometimes see this error during downloads:

"Exception caught: Maximum retry attempts reached. Unable to get a valid decryption key."

And I've noticed sometimes paid content is not being downloaded by the app.  Can I take this to mean that, for some reason, authentication has failed mid-download?

Thanks!

**Comment by melithine** [comment]
Are you using your own CDM/Widevine keys, or are you relying on cdrm-project? (There's a yellow message at startup if the latter.)

**Comment by Smackdab69** [comment]
device_client_id_blob and/or device_private_key missing, https://cdrm-project.com/ will be used instead for DRM
protected videos

So I assume I'm using CDRM?

**Comment by melithine** [comment]
Yes. That's probably the problem; cdrm-project has frequent outages. If you hit up in the Discord server, I or one of the other folks there can give you keys, which should probably solve the problem.
